File: tournaments/chessKnightMoves/chessKnightMoves.js
function chessKnightMoves(cell) {
    const a = cell[0].charCodeAt() - 'a'.charCodeAt();
    const b = cell[1].charCodeAt() - '0'.charCodeAt();
    let result = 0;

    if (0 <= a - 1 && 1 <= b - 2) ++result;
    if (0 <= a - 2 && 1 <= b - 1) ++result;
    if (7 >= a + 2 && 1 <= b - 1) ++result;
    if (7 >= a + 1 && 1 <= b - 2) ++result;
    if (0 <= a - 1 && 8 >= b + 2) ++result;
    if (0 <= a - 2 && 8 >= b + 1) ++result;
    if (7 >= a + 1 && 8 >= b + 2) ++result;
    if (7 >= a + 2 && 8 >= b + 1) ++result;

    return result;
}

File: tournaments/chessKnightMoves/chessKnightMoves.cpp
int chessKnightMoves(std::string cell) {

  struct Helper {
    bool isValid(int pos) {
      if (0 <= pos && pos < 8) {
        return true;
      }
      return false;
    }

    int getX(std::string pos) {
      return pos[0] - 'a';
    }

    int getY(std::string pos) {
      return pos[0] - '1';
    }
  };
  Helper h;

  int current_x = h.getX(cell.substr(0, 1));
  int current_y = h.getY(cell.substr(1, 2));
  int result = 0;

  for (int dx = -2; dx <= 2; dx++) {
    for (int dy = -2; dy <= 2; dy++) {
      if (std::abs(dx * dy) == 2) {
        if (h.isValid(current_x + dx) && h.isValid(current_y + dy)) {
          result++;
        }
      }
    }
  }
  return result;

}
